File: packages/cpp_talker/src/talker.cpp
#include <chrono>
#include <memory>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

#include "cpp_greeter/greeter.hpp"

using namespace std::chrono_literals;

class Talker : public rclcpp::Node
{
public:
  Talker()
  : Node("cpp_talker"), count_(0)
  {
    publisher_ = this->create_publisher<std_msgs::msg::String>("demo_chatter", 10);
    timer_ = this->create_wall_timer(
      500ms,
      [this]() {
        std_msgs::msg::String msg;
        msg.data = cpp_greeter::make_greeting(count_++);
        RCLCPP_INFO(this->get_logger(), "publishing: %s", msg.data.c_str());
        publisher_->publish(msg);
      });
  }

private:
  rclcpp::TimerBase::SharedPtr timer_;
  rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
  int count_;
};

int main(int argc, char ** argv)
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<Talker>());
  rclcpp::shutdown();
  return 0;
}

File: packages/py_talker/py_talker/talker.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class Talker(Node):
    def __init__(self):
        super().__init__("py_talker")
        self.publisher = self.create_publisher(String, "demo_chatter", 10)
        self.count = 0
        self.create_timer(0.5, self.tick)

    def tick(self):
        msg = String(data=f"hello-from-py-talker {self.count}")
        self.publisher.publish(msg)
        self.get_logger().info(f"publishing: {msg.data}")
        self.count += 1


def main():
    rclpy.init()
    node = Talker()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()


if __name__ == "__main__":
    main()
